plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergSplitSource.java

```diff
@@ -256,6 +266,24 @@ else if (upperBound != null) {
         return Domain.create(ValueSet.ofRanges(statisticsRange), containsNulls);
     }
 
+    @VisibleForTesting
+    static boolean partitionMatchesConstraint(
+            Set<IcebergColumnHandle> identityPartitionColumns,
+            Map<Integer, Optional<String>> partitionKeys,
+            ConstraintEvaluator constraintEvaluator)
+    {
+        Map<ColumnHandle, NullableValue> bindings = new HashMap<>();
+        for (IcebergColumnHandle partitionColumn : identityPartitionColumns) {
+            Object partitionValue = deserializePartitionValue(
```

Comment:
We already maybe have called `deserializePartitionValue` for  DF purposes.

Maybe we should have it as

```java
Supplier<Map<IcebergColumnHandle, NullableValue >> partitionValues = memoize(() -> {
  // deserializePartitionValue for each identityPartitionColumns
});
```

Reply:
bump?